figure_generation/01_figure_generation.py

```diff
@@ -0,0 +1,364 @@
+        color="Relation (LF) Source",
+        x="Number of Label Functions Added",
+        y="AUPR",
+    )
+)
+g.save("output/figure_five.svg")
+g.save("output/figure_five.png", dpi=300)
+print(g)
+
+# # Discriminator Model Figures
+
+# ## Figure 5 - Discriminator Model vs Generative Model
+
+disc_performance_df = pd.read_csv(
+    Path("../discriminative_model_training/output/all_total_lf_performance.tsv"),
+    sep="\t",
+)
+disc_performance_df.lf_num = pd.Categorical(
+    disc_performance_df.lf_num.tolist(), categories=["0", "1", "6", "11", "16", "All"]
+)
+disc_performance_df
+
+g = (
+    p9.ggplot(disc_performance_df >> ply.query("dataset=='test'"))
+    + p9.aes(
+        x="lf_num",
+        y="auroc_mean",
+        ymin="auroc_lower_ci",
+        ymax="auroc_upper_ci",
+        group="model",
+        color="label_source",
+        linetype="model",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.scale_color_manual(values=color_map)
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.labs(
+        title="Discriminative Model Performance (Test Set/AUROC)",
+        linetype="Model",
+        color="Relation (LF) Source",
+        x="Number of Label Functions Added",
+        y="AUROC",
+    )
+    + p9.theme(figure_size=(8, 6))
+)
+g.save("output/figure_six.svg")
+g.save("output/figure_six.png", dpi=300)
+print(g)
+
+g = (
+    p9.ggplot(disc_performance_df >> ply.query("dataset=='test'"))
+    + p9.aes(
+        x="lf_num",
+        y="aupr_mean",
+        ymin="aupr_lower_ci",
+        ymax="aupr_upper_ci",
+        group="model",
+        color="label_source",
+        linetype="model",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.scale_color_manual(values=color_map)
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.labs(
+        title="Discriminative Model Performance (Test Set/AUPR)",
+        linetype="Model",
+        color="Relation (LF) Source",
+        x="Number of Label Functions Added",
+        y="AUPR",
+    )
+    + p9.theme(figure_size=(8, 6))
+)
+g.save("output/figure_seven.svg")
+g.save("output/figure_seven.png", dpi=300)
+print(g)
+
+# # Edge Prediction
+
+edge_recall_df = (
+    pd.concat(
+        [
+            pd.read_csv(recall_file_path, sep="\t")
+            for recall_file_path in Path("../edge_prediction_experiment/output/").rglob(
+                "*edge_recall.tsv"
+            )
+        ]
+    )
+    >> ply.call(".reset_index")
+    >> ply.select("-index")
+)
+edge_recall_df
+
+g = (
+    p9.ggplot(edge_recall_df, p9.aes(x="relation", y="edges", fill="in_hetionet"))
+    + p9.geom_col(position="dodge")
+    + p9.geom_text(
+        p9.aes(
+            label=(
+                edge_recall_df.apply(
+                    lambda x: f"{x['edges']:,}\n({x['recall']:.0%})"
+                    if not math.isnan(x["recall"])
```

Comment:
When would the recall be `nan`? Does this just ignore those cases? This seems like it might be worth a short comment to me, but it's up to you.

Reply:
>When would the recall be nan? Does this just ignore those cases? This seems like it might be worth a short comment to me, but it's up to you.

Nan in this case are for the new edges that could be added to "Hetionet". Recall doesn't make sense in this case because you cannot know what the ground truth in this case.